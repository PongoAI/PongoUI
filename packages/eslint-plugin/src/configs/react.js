/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
