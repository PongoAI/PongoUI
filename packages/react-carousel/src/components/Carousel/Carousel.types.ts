import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
import type { RadioGroup } from '@cebus/react-radio';
import React from 'react';

export type CarouselImage = React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>;

export type CarouselChange = React.MouseEvent<HTMLElement> | React.FormEvent<HTMLDivElement>;

export type CarouselSlots = {
  /**
   * The root of the Carousel.
   */
  root: Slot<'div'>;

  /**
   * The tabpanel wrapper around the images.
   */
  tabpanel: NonNullable<Slot<'div'>>;

  /**
   * The back button for the Carousel
   */
  backArrow?: Slot<'button'>;

  /**
   * The forward button for the Carousel
   */
  forwardArrow?: Slot<'button'>;

  /**
   * The wrapper for the radio button indicators in the minimal appearance.
   */
  radioIndicator?: Slot<typeof RadioGroup>;
};

type CarouselCommons = {
  /**
   * The uncontrolled index of the currently visible page.
   */
  defaultValue?: number;

  /**
   * The controlled index of the currently visible page.
   */
  value?: number;

  /**
   * The callback to call when the Carousel changes.
   */
  onChange?: (ev: CarouselChange, data: { value: number }) => void;

  /**
   * The appearance of the Carousel.
   *
   * * minimal: Arrows are shown to the left and right of the image and a bar indicator shows the position.
   * * gallery: No arrows are visible and the user is shown their location by a paginated set of small images.
   */
  appearance?: 'minimal' | 'gallery';

  /**
   * The content to render within the Carousel
   */
  content: {
    /**
     * The image source to load.
     */
    src: string;

    /**
     * The alt text to be read by the screen reader
     */
    alt: string;
  }[];
};

export type CarouselProps = Omit<ComponentProps<Partial<CarouselSlots>>, 'children'> & {
  /**
   * Add images through the content prop
   */
  children?: never;
} & CarouselCommons;

export type CarouselState = ComponentState<CarouselSlots> & CarouselCommons;
