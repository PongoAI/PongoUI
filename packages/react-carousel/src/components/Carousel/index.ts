export * from './Carousel';
export * from './useCarouselStyles';
