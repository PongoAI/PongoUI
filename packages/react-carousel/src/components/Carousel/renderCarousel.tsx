import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { CarouselState, CarouselSlots } from './Carousel.types';

export const renderCarousel = (state: CarouselState) => {
  const { slots, slotProps } = getSlots<CarouselSlots>(state);

  return (
    <slots.root {...slotProps.root}>
      <slots.tabpanel {...slotProps.tabpanel}>{state.root.children}</slots.tabpanel>
      {slots.backArrow && state.appearance === 'minimal' && <slots.backArrow {...slotProps.backArrow} />}
      {slots.forwardArrow && state.appearance === 'minimal' && <slots.forwardArrow {...slotProps.forwardArrow} />}
      {slots.radioIndicator && state.appearance === 'minimal' && <slots.radioIndicator {...slotProps.radioIndicator} />}
    </slots.root>
  );
};
