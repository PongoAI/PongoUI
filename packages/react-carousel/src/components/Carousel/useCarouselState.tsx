import * as React from 'react';
import { Radio } from '@cebus/react-radio';
import { ForwardArrowIcon, BackArrowIcon } from '@cebus/react-icon';
import { useControllableState, useEventCallback } from '@fluentui/react-utilities';
import { tokens } from '@cebus/react-theme';
import type { CarouselState, CarouselChange } from './Carousel.types';
import type { RadioGroupProps } from '@cebus/react-radio';

const imageStyles: React.CSSProperties = {
  position: 'relative',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  userSelect: 'none',
};

export const useCarouselState = (state: CarouselState): CarouselState => {
  const { content, appearance, defaultValue, value, onChange } = state;

  const [currentValue, setCurrentValue] = useControllableState({
    defaultState: defaultValue,
    state: value,
    initialState: 0,
  });

  const onImageChange = useEventCallback((ev: CarouselChange, incomingValue: number) => {
    // If we go past the first item
    const clampedValue =
      incomingValue < 0 ? content.length - 1 : incomingValue > content.length - 1 ? 0 : incomingValue;
    onChange?.(ev, { value: clampedValue });
    setCurrentValue(clampedValue);
  });

  const calculateTransform = () => currentValue * 100;

  const imageWrapperStyles = {
    width: '100%',
    height: '100%',
    flexShrink: '0',
    transform: `translateX(-${calculateTransform()}%)`,
    transitionProperty: 'transform',
    transitionDuration: tokens.transitionDuration,
    transitionTimingFunction: tokens.transitionTimingFunction,
  };

  // const childrenWithProps = React.Children.map(children, child => {
  //   if (React.isValidElement(child)) {
  //     return (
  // <div style={imageWrapperStyles}>
  //   {React.cloneElement(child, {
  //     draggable: false,
  //     style: imageStyles,
  //   })}
  // </div>
  //     );
  //   }
  //   return child;
  // });

  const createImages = content.map(item => {
    return (
      <div key={content.indexOf(item)} style={imageWrapperStyles}>
        <img src={item.src} alt={item.alt} draggable={false} style={imageStyles} />
      </div>
    );
  });

  const onRadioChange: RadioGroupProps['onChange'] = (ev, data) => {
    onImageChange(ev, Number.parseInt(data.value));
  };

  const onBackButtonClick = (ev: React.MouseEvent<HTMLElement>) => {
    onImageChange(ev, currentValue - 1);
  };

  const onForwardButtonClick = (ev: React.MouseEvent<HTMLElement>) => {
    onImageChange(ev, currentValue + 1);
  };

  state.root.children = createImages;

  const createRadioButtons = () => {
    const buttons = [];

    for (let i = 0; i < content.length; i++) {
      buttons.push(
        <Radio key={i} role="carousel" value={i.toString()} aria-label={content[currentValue].alt} readOnly />,
      );
    }

    return buttons;
  };

  if (appearance === 'minimal' && state.backArrow) {
    state.backArrow.children = <BackArrowIcon />;
    state.backArrow.onClick = onBackButtonClick;
  }

  if (appearance === 'minimal' && state.forwardArrow) {
    state.forwardArrow.children = <ForwardArrowIcon />;
    state.forwardArrow.onClick = onForwardButtonClick;
  }

  if (appearance === 'minimal' && state.radioIndicator) {
    state.radioIndicator.value = currentValue.toString();
    state.radioIndicator.onChange = onRadioChange;
    state.radioIndicator.children = createRadioButtons();
  }

  return state;
};
