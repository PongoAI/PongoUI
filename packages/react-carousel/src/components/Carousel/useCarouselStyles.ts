import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { CarouselState } from './Carousel.types';
import { tokens } from '@pongo-ui/react-theme';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    display: 'inline-flex',
    ...shorthands.border('1px', 'solid', tokens.inheritBackground),
    ...shorthands.borderRadius(tokens.rounded),
    flexDirection: 'row',
  },

  tabpanel: {
    position: 'relative',
    display: 'inline-flex',
    alignItems: 'center',
    ...shorthands.overflow('hidden'),
    outlineStyle: 'none',
  },

  arrow: {
    position: 'absolute',
    width: '34px',
    height: '34px',
    flexShrink: 0,
    flexGrow: 0,
    ...shorthands.borderStyle('none'),
    opacity: '20%',
    ...shorthands.borderRadius(tokens.circle),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    filter: tokens.elevate,
    backgroundColor: tokens.canvasColor,
    cursor: 'pointer',
    outlineStyle: 'none',
    ...shorthands.margin(tokens.focusedLayout),
    top: '50%',
    ':hover': {
      opacity: '50%',
    },
    ':active': {
      opacity: '60%',
    },
  },

  backArrow: {
    left: '0px',
  },

  forwardArrow: {
    right: '0px',
  },

  radioIndicator: {
    position: 'absolute',
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    boxSizing: 'border-box',
    ...shorthands.padding(tokens.focusedLayout),
    justifyContent: 'center',
    bottom: '0px',
    backgroundColor: tokens.canvasColor,
    opacity: '70%',
    width: '100%',
  },
});

export const useCarouselStyles = (state: CarouselState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(styles.root, state.root.className);

  state.tabpanel.className = mergeClasses(styles.tabpanel, state.tabpanel.className);

  if (state.backArrow) {
    state.backArrow.className = mergeClasses(styles.arrow, styles.backArrow, state.backArrow.className);
  }

  if (state.forwardArrow) {
    state.forwardArrow.className = mergeClasses(styles.arrow, styles.forwardArrow, state.forwardArrow.className);
  }

  if (state.radioIndicator) {
    state.radioIndicator.className = mergeClasses(styles.radioIndicator, state.radioIndicator.className);
  }

  return state;
};
