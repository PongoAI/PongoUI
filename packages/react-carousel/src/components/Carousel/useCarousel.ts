import * as React from 'react';
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
import { useCarouselState } from './useCarouselState';
import { RadioGroup } from '@pongo-ui/react-radio';
import type { CarouselProps, CarouselState } from './Carousel.types';

export const useCarousel = (props: CarouselProps, ref: React.Ref<HTMLElement>): CarouselState => {
  const {
    appearance = 'minimal',
    defaultValue = 0,
    value,
    onChange,
    tabpanel,
    content,
    backArrow,
    forwardArrow,
    radioIndicator,
  } = props;

  const state: CarouselState = {
    defaultValue,
    value,
    onChange,
    content,
    appearance,
    components: {
      root: 'div',
      tabpanel: 'div',
      backArrow: 'button',
      forwardArrow: 'button',
      radioIndicator: RadioGroup,
    },
    root: getNativeElementProps('div', {
      ...props,
      ref,
    }),
    tabpanel: resolveShorthand(tabpanel, {
      required: true,
      defaultProps: {
        role: 'tabpanel',
      },
    }),
    radioIndicator: resolveShorthand(radioIndicator, {
      required: true,
      defaultProps: { layout: 'horizontal' },
    }),
    backArrow: resolveShorthand(backArrow, {
      required: true,
      defaultProps: {
        type: 'button',
        tabIndex: -1,
      },
    }),
    forwardArrow: resolveShorthand(forwardArrow, {
      required: true,
      defaultProps: {
        type: 'button',
        tabIndex: -1,
      },
    }),
  };

  useCarouselState(state);

  return state;
};
