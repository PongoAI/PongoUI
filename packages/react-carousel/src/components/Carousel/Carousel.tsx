import * as React from 'react';
import { useCarousel } from './useCarousel';
import { renderCarousel } from './renderCarousel';
import { useCarouselStyles } from './useCarouselStyles';
import type { CarouselProps } from './Carousel.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Carousel component is used to display a collection of images.
 */
export const Carousel: ForwardRefComponent<CarouselProps> = React.forwardRef((props, ref) => {
  const state = useCarousel(props, ref);

  useCarouselStyles(state);

  return renderCarousel(state);
});
Carousel.displayName = 'Carousel';
