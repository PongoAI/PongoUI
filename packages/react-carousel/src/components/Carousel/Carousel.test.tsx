import * as React from 'react';
import { Carousel } from './Carousel';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Carousel', () => {
  isConformant({
    Component: Carousel,
    displayName: 'Carousel',
  });

  describe('Snapshot tests', () => {
    it('renders a basic Avatar', () => {
      const result = render(<Carousel />);

      expect(result.baseElement).toMatchSnapshot();
    });
  });
});
