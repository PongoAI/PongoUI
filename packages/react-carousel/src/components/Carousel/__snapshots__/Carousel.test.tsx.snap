// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Snapshot tests renders a basic Avatar 1`] = `
<body>
  <div>
    <div
      class="___abcusc0 f10pi13n ftuwxu6 f5ogflp f1hqa2wf f1f09k3d finvdd3 fzkkow9 fcdblym fg706s2 fjik90z f17iqts3 f1uoix4j f18idrzn f8q3d83 f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1063pyq"
    >
      <div
        class="___btqw720 f10pi13n ftuwxu6 f122n59 f1p9o1ba f1sil6mw f1s6fcnf"
      >
        <div
          style="width: 100%; height: 100%; flex-shrink: 0; transform: translateX(-0%); transition-property: transform; transition-duration: var(--transitionDuration); transition-timing-function: var(--transitionTimingFunction);"
        >
          <img
            alt="A white t shirt"
            draggable="false"
            src="https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/tshirts-1613587154.jpg"
            style="position: relative; width: 100%; height: 100%; object-fit: contain; user-select: none;"
          />
        </div>
        <div
          style="width: 100%; height: 100%; flex-shrink: 0; transform: translateX(-0%); transition-property: transform; transition-duration: var(--transitionDuration); transition-timing-function: var(--transitionTimingFunction);"
        >
          <img
            alt="A green t shirt"
            draggable="false"
            src="https://imgprd19.hobbylobby.com/2/4f/57/24f57e245a879cb2543edd1df4e090bfebf24a45/700Wx700H-1013689-0320.jpg"
            style="position: relative; width: 100%; height: 100%; object-fit: contain; user-select: none;"
          />
        </div>
      </div>
      <button
        class="___3hn4950 f1euv43f f5mk7bh fbsu25e fi64zpg f1jhi6b8 f1ern45e f1n71otn f1h8hb77 f1deefiw fb1ohyo fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f22iagw f122n59 f4d9j23 fl4n2cu f8qjfyk f1k6fduh f1s6fcnf f1btex71 fr1s5rp f3lqehk fcm09hb f1i1t8d1 f1liqzi9 f1p7xwk3 f1xynx9j"
        tabindex="-1"
        type="button"
      >
        <svg
          class="___gj5ft00 f14t3ns0 fmrv4ls f6qemis fq4mcun frvgh55"
          focusable="false"
          style="--cebus-icon-fill: var(--textColor);"
          viewBox="0 0 2048 2048"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="160"
            rx="80"
            transform="rotate(43.1013 692.688 932.496)"
            width="1058.19"
            x="692.688"
            y="932.496"
          />
          <rect
            height="160"
            rx="80"
            transform="rotate(133.101 1423.18 384.326)"
            width="1058.63"
            x="1423.18"
            y="384.326"
          />
        </svg>
      </button>
      <button
        class="___1l6nr4d f1euv43f f5mk7bh fbsu25e fi64zpg f1jhi6b8 f1ern45e f1n71otn f1h8hb77 f1deefiw fb1ohyo fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f22iagw f122n59 f4d9j23 fl4n2cu f8qjfyk f1k6fduh f1s6fcnf f1btex71 fr1s5rp f3lqehk fcm09hb f1i1t8d1 f1liqzi9 f1p7xwk3 f10k790i"
        tabindex="-1"
        type="button"
      >
        <svg
          class="___gj5ft00 f14t3ns0 fmrv4ls f6qemis fq4mcun frvgh55"
          focusable="false"
          style="--cebus-icon-fill: var(--textColor);"
          viewBox="0 0 2048 2048"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="160"
            rx="80"
            transform="rotate(-132.67 1359.52 1129.17)"
            width="1058.19"
            x="1359.52"
            y="1129.17"
          />
          <rect
            height="160"
            rx="80"
            transform="rotate(-42.6701 590.599 1621.98)"
            width="1058.63"
            x="590.599"
            y="1621.98"
          />
        </svg>
      </button>
      <div
        class="___oo1c720 f22iagw f122n59 fnmxc1e fynvm4a f1euv43f fni485r f1ewtqcl fk7qxxv f10sqwl7 f1eg49sl fne32i3 f4d9j23 f5gq2j6 f8qjfyk f12qha5f fly5x3f"
        role="radiogroup"
      >
        <span
          class="___18rr115 ftuwxu6 f10pi13n fnmxc1e f122n59 f1ewtqcl fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f8qjfyk f2hkw1w fdxsi5y f1i4q40k fz5pyj9 fuitljl f1lxa25c f1dexcea fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fgm4zzz f1o1s7a4 f1w999bl fr2bnm2 f1jjwxw3 fijdmwi f11pcu7g f1ddvjn"
        >
          <input
            aria-label="A white t shirt"
            checked=""
            class="___5a0q0l0 f1euv43f f1vgc2s3 f15twtuk f1e31b4d f1yab3r1 fly5x3f f1l02sjl f1ewtqcl fk73vx1 f1hu3pq6 f11qmguv f19f4twv f1tyq0we f9q6s5f f1bjzxfu f1k6fduh fbqqmqa f12q32tx fvhvjdh fsk3gdh fkx5us4 f4sr7pu fy8k3os f148sgik f1erntv0 flba4d6 f1yjaxmd fhnq91 f1az57 f18hx1k2 fiyrlox fc7evkt fdh8j9a f16pqu9g fv8ay8h fddmz4c f19ixr9n f1ktf4xn fojjxu7 fjyby1k f8z8274 fp62gv2 f4c4h6k fxkwkjs f27w35r f107t6ws f1umzgs4"
            id="radio-14"
            name="radiogroup-13"
            readonly=""
            role="carousel"
            type="radio"
            value="0"
          />
          <div
            aria-hidden="true"
            class="cebus-radio-indicator ___n55l230 f22iagw fjw5fx7 fd461yt f1ewtqcl fi64zpg f122n59 f4d9j23 f1aehjj5 f1p9o1ba f1sil6mw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f19qt5od ftovyd4 f1fajnbs ffz4t7b f1wac4k f1t8un4"
          >
            <svg
              aria-hidden="true"
              class="___1yjlmar f1w7gpdv fez10in f1kxrtqx"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 2a8 8 0 100 16 8 8 0 000-16z"
              />
            </svg>
          </div>
        </span>
        <span
          class="___18rr115 ftuwxu6 f10pi13n fnmxc1e f122n59 f1ewtqcl fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f8qjfyk f2hkw1w fdxsi5y f1i4q40k fz5pyj9 fuitljl f1lxa25c f1dexcea fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fgm4zzz f1o1s7a4 f1w999bl fr2bnm2 f1jjwxw3 fijdmwi f11pcu7g f1ddvjn"
        >
          <input
            aria-label="A white t shirt"
            class="___5a0q0l0 f1euv43f f1vgc2s3 f15twtuk f1e31b4d f1yab3r1 fly5x3f f1l02sjl f1ewtqcl fk73vx1 f1hu3pq6 f11qmguv f19f4twv f1tyq0we f9q6s5f f1bjzxfu f1k6fduh fbqqmqa f12q32tx fvhvjdh fsk3gdh fkx5us4 f4sr7pu fy8k3os f148sgik f1erntv0 flba4d6 f1yjaxmd fhnq91 f1az57 f18hx1k2 fiyrlox fc7evkt fdh8j9a f16pqu9g fv8ay8h fddmz4c f19ixr9n f1ktf4xn fojjxu7 fjyby1k f8z8274 fp62gv2 f4c4h6k fxkwkjs f27w35r f107t6ws f1umzgs4"
            id="radio-15"
            name="radiogroup-13"
            readonly=""
            role="carousel"
            type="radio"
            value="1"
          />
          <div
            aria-hidden="true"
            class="cebus-radio-indicator ___n55l230 f22iagw fjw5fx7 fd461yt f1ewtqcl fi64zpg f122n59 f4d9j23 f1aehjj5 f1p9o1ba f1sil6mw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fxvwsu1 f3ia6t0 f1e3jql2 f1gxgi0c f19qt5od ftovyd4 f1fajnbs ffz4t7b f1wac4k f1t8un4"
          >
            <svg
              aria-hidden="true"
              class="___1yjlmar f1w7gpdv fez10in f1kxrtqx"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 2a8 8 0 100 16 8 8 0 000-16z"
              />
            </svg>
          </div>
        </span>
      </div>
    </div>
  </div>
</body>
`;
